src/core/ext/xds/certificate_provider_store.cc

```diff
@@ -49,14 +50,21 @@ CertificateProviderStore::CreateOrGetCertificateProvider(
   }
   RefCountedPtr<grpc_tls_certificate_provider> cert_provider =
       factory->CreateCertificateProvider(plugin_config_it->second.config);
-  certificate_providers_map_.insert({plugin_config_it->first, cert_provider});
+  certificate_providers_map_.emplace(
```

Comment:
Note that this needs to be this complicated since grpc_core::RefCount is not copyable or movable.